### ISSUE TITLE
fair(k>=2) does not converge with cotangent weights (the default)

### ISSUE THREAD

**Issue by KhoiDOO** [KhoiDOO/conquer3d#63]
`TriangleMesh.fair()` runs the order-$k$ flow to a steady state. With **cotangent weights
(`mode=1`) and $k \ge 2$ it never reaches one** — the surface keeps moving indefinitely
instead of settling. Since the defaults are `k=2, mode=1`, the default configuration is the
failing one.

`smooth()` and `fair(k=1)` with the same weights are fine, so the cotangent machinery
itself works; it is the $k$-fold composition that breaks.

## Reproduction

```python
import torch
from conquer3d.creation import create_sphere
from conquer3d.data_structure import TriangleMesh

sv, sf = create_sphere(sectors=160, stacks=80)[:2]
sv, sf = sv.cuda().float().contiguous(), sf.cuda().int().contiguous()

mesh = TriangleMesh(sv.clone(), sf.clone())
prev = mesh.vertices.clone()
for _ in range(4):
    mesh.fair(k=2, iterations=5000, damping=0.25, mode=1)   # defaults but for damping
    v = mesh.vertices
    print(f"{float((v - prev).norm(dim=-1).max()):.2e}")    # never decays
    prev = v.clone()
```

Movement per 5,000-iteration chunk should decay toward zero. It does not.

## Behaviour matrix

Generated UV sphere, 12,642 vertices, `damping = 0.5 * 2**(1-k)`, four chunks of 5,000
iterations. The number shown is the max per-vertex movement in each chunk.

| k | mode | locked | chunk-to-chunk movement | verdict |
|---|---|---|---|---|
| 1 | cotangent | yes | 2.8e-1 4.5e-2 5.6e-2 5.7e-2 | converging |
| **2** | **cotangent** | yes | 3.5e-2 4.2e-2 3.9e-2 3.0e-2 | **oscillating** |
| **3** | **cotangent** | yes | 2.3e-2 2.2e-2 2.7e-2 2.7e-2 | **oscillating** |
| 1 | uniform | yes | 1.9e-1 2.5e-5 7.5e-9 2.0e-9 | converging |
| 2 | uniform | yes | 2.8e-2 6.1e-3 3.8e-3 2.4e-3 | converging |
| 3 | uniform | yes | 1.7e-2 2.2e-3 1.4e-3 1.1e-3 | converging |
| **2** | **cotangent** | no | 3.7e-2 3.3e-2 3.5e-2 5.2e-2 | **oscillating** |
| **3** | **cotangent** | no | 2.3e-2 2.4e-2 2.7e-2 3.4e-2 | **oscillating** |
| 2 | uniform | no | 2.8e-2 5.6e-3 4.0e-3 3.3e-3 | converging |
| 3 | uniform | no | 1.7e-2 2.2e-3 1.3e-3 9.8e-4 | converging |

Perfectly separated by `mode`. The `locked` mask makes no difference.

## It is not the step size

Igea, `k=2`, `mode=1`. The documented stability bound is $2^{1-k} = 0.5$.

| damping | chunk-to-chunk movement | total displacement | patch mean \|H\| |
|---|---|---|---|
| 0.250 | 1.3e-2 1.2e-2 1.1e-2 1.1e-2 | 0.0172 | 4.030 |
| 0.100 | 9.4e-3 1.1e-2 7.9e-3 6.9e-3 | 0.0115 | 3.638 |
| 0.050 | 8.0e-3 6.3e-3 6.9e-3 5.6e-3 | 0.0103 | 3.422 |
| 0.020 | 5.3e-3 5.6e-3 3.8e-3 5.4e-3 | 0.0091 | 3.362 |
| 0.005 | 4.7e-3 3.0e-3 1.6e-3 2.6e-3 | 0.0053 | 3.674 |

At 1/100th of the bound it still does not decay. Note the total displacement *stalls* around
0.01-0.02 while per-chunk movement stays flat: vertices keep moving without going anywhere,
which is a limit cycle rather than slow progress. Mean \|H\| inside the patch also *rises*
with more iterations, when a thin-plate flow should reduce it.

For contrast, `mode=0` on the same mesh decays monotonically: `1.3e-2 4.4e-3 2.6e-3 1.8e-3`.

## Suspected cause

Cotangent weights go negative on obtuse triangles. Both test meshes have plenty — Igea
19.6%, the UV sphere 49.4% (pole slivers).

A single application of $\Delta$ tolerates this: the operator is weight-sum normalised and
still averages. $\Delta^k$ for $k \ge 2$ does not, because composing a non-sign-definite
operator with itself loses the spectral property the explicit Euler step relies on. Modes
with negative effective eigenvalues are then amplified rather than damped, and no step size
fixes that — which is exactly what the damping sweep shows.

Worth checking in `fair_laplacian` (`triangle_mesh.cu`):

1. whether cotangent weights are rebuilt *between* the $k$ applications within one step,
   making the composed operator time-varying rather than a true $\Delta^k$;
2. whether the $(-1)^{k+1}$ sign is applied once per step or once per application;
3. whether clamping negative cotangent weights to zero (a standard remedy) restores
   convergence while keeping the geometry-following behaviour cotangent weights are chosen for.

## Impact

- The default call `mesh.fair()` does not converge.
- The docstring states cotangent is "strongly preferred here", so the recommended
  configuration is the broken one, while the one that works (`mode=0`) is documented as
  leaving artifacts where vertex density varies.
- Blocks the fairing showcase: a figure at a fixed iteration count would present an
  oscillating intermediate state as the thin-plate and minimum-variation surfaces.

**Comment by KhoiDOO** [comment]
A third reproduction case, from building the fairing showcase — and it **rules out my
original hypothesis**.

## The generated tube diverges, it does not merely oscillate

A swept elbow (4,416 verts, 8,736 faces), straight ends locked, bend free. Four chunks of
2,000 iterations, `damping = 0.5 * 2**(1-k)`, showing max per-vertex movement per chunk:

| k | mode | movement per chunk | verdict |
|---|---|---|---|
| 1 | cotangent | 3.5e-1 6.6e-2 4.2e-2 4.4e-2 4.8e-2 | stalls |
| **2** | **cotangent** | 7.5e-3 2.3e-2 3.7e-2 **7.4e-2** 4.1e-2 | **diverging** |
| **3** | **cotangent** | 7.1e-4 1.4e-2 2.3e-2 3.2e-2 3.2e-2 | **diverging** |
| 1 | uniform | 4.1e-1 6.5e-3 2.1e-4 8.6e-7 7.5e-9 | converging |
| 2 | uniform | 1.7e-2 9.8e-3 8.1e-3 7.3e-3 6.8e-3 | converging, slowly |
| 3 | uniform | 6.8e-3 3.1e-3 2.3e-3 1.9e-3 1.6e-3 | converging, slowly |

Note k=2 cotangent: movement **grows** by an order of magnitude. On Igea and the sphere it
held a limit cycle; here it runs away.

## My obtuse-triangle hypothesis does not hold

I built this tube deliberately to be well-triangulated — ring spacing matched to
circumferential spacing, so the quads are square. It still comes out **15.9% obtuse**
(the quad diagonal makes near-right triangles), and cotangent still fails.

So obtuse triangles remain a plausible contributor but are **not established as the cause**.
All three meshes tested have a substantial obtuse fraction, so the experiment does not
separate the hypotheses. A mesh with near-zero obtuse triangles — an icosphere, or a
Delaunay-remeshed patch — would be the discriminating test, and I have not run it.

What is solid: the failure tracks `mode`, not mesh provenance, not the `locked` mask, and
not the step size.

## Uniform weights converge, but slowly, and that part is expected

Running to a 1e-6 tolerance on the tube:

| k | iterations to steady state | final movement | displacement |
|---|---|---|---|
| 1 | 50,000 | 3.7e-9 | 0.417 |
| 2 | 400,000 | 3.6e-7 | 0.143 |
| 3 | 4,000,000 | 2.3e-4 (asymptotic) | 0.103 |

Roughly 10x more steps per order, which is the stiffness the docstring already warns about
rather than a second defect.

## Consequence for the docs

The fairing showcase (#66) ships with `mode=0` for exactly this reason. It is the only mode
that reaches a steady state, so it is the only one whose output can honestly be labelled a
thin-plate or minimum-variation surface. That means the gallery demonstrates a
non-default, non-recommended configuration, and will keep doing so until cotangent
converges.
